generate_qt_mappings.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+"""
+This script generated the Qt mapping file according to given Qt include
+directory
+
+Example usage :
+
+   $ ./generate_qt_mappings.py /usr/include/x86_64-linux-gnu/qt5 qt5_11.imp
+
+"""
+
+import argparse
+import glob
+import os
+import re
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("qt_include_dir", help="Qt include directoy")
+    parser.add_argument("output_file", help="Generated output mapping file")
+    parser.add_argument("--warnings", action="store_true", help="Display possible warning messages")
+    args = parser.parse_args()
+
+    headers = glob.glob(os.path.join(args.qt_include_dir, '**/*[!\\.h]'))
+    symbols_map = []
+    includes_map = []
+
+    class2includes = {}
+    include2classes = {}
+    for header in headers:
+
+        if os.path.isdir(header):
+            continue
+
+        class_name = os.path.basename(header)
+        module_name = os.path.basename(os.path.dirname(header))
+
+        symbols_map += [f'{{ symbol: [ "{class_name}", "private", '
+                        f'"<{class_name}>", "public" ] }}']
+
+        content = open(header, 'r').read()
+        includes = re.findall(r'#include "(.*)\.h"', content)
+        if len(includes) != 1:
+            continue
+
+        include = includes[0]
+        class2includes[class_name] = (class2includes.get(class_name, [])
+                                      + [header])
+
+        full_include = module_name + "/" + include
+        include2classes[full_include] = (include2classes.get(full_include, [])
+                                         + [class_name])
+
+        includes_map += [
+            f'{{ include: [ "@[\\"<]({module_name}/)?{include}\\\\.h[\\">]", '
+            f'"private", "<{class_name}>", "public" ] }}']
+
+    # Print some warnings
+    if args.warnings:
+        class2includes = [(class_name, includes) for class_name, includes in
+                          class2includes.items() if len(includes) > 1]
+
+        if class2includes:
+            print("Warning! These classes references several includes file:")
+            for class_name, includes in class2includes:
+                print("  {} : {}".format(class_name, ",".join(includes)))
+
+        include2classes = [(include, classes) for include, classes in
+                           include2classes.items() if len(classes) > 1]
+        if include2classes:
+            print("Warning! These includes references several classes:")
+            for include, classes in include2classes:
+                print("  {} : {}".format(include, ",".join(classes)))
+
+    f = open(args.output_file, 'w')
```

Comment:
with open(args.output_file, 'w') as f:
        ...

I'd also use the Python3 print function (with `from __future__ import print_function` at the top of the file):

    print("# This file has been generated by the script {}. Do not edit!".format(
            os.path.basename(__file__)), file=f)

and then use the `json` module to serialize the record list.